samples/web/content/apprtc/js/roomselection.js

```diff
@@ -0,0 +1,139 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+/* More information about these options at jshint.com/docs/options */
+
+/* globals randomString, Storage, parseJSON */
+/* exported RoomSelection */
+
+'use strict';
+
+var RoomSelection = function(roomSelectionDiv, uiConstants, recentRoomsKey) {
+  this.roomSelectionDiv_ = roomSelectionDiv;
+  
+  this.uiConstants = uiConstants;
+  
+  this.roomIdInput_ = this.roomSelectionDiv_.querySelector(this.uiConstants.roomSelectionInput);
+  this.roomJoinButton_ = this.roomSelectionDiv_.querySelector(this.uiConstants.roomSelectionJoinButton);
+  this.roomRandomButton_ = this.roomSelectionDiv_.querySelector(this.uiConstants.roomSelectionRandomButton);
+  this.roomRecentList_ = this.roomSelectionDiv_.querySelector(this.uiConstants.roomSelectionRecentList);
+  
+  this.roomIdInput_.value = randomString(9);
+  // Call onRoomIdInput_ now to validate initial state of input box.
+  this.onRoomIdInput_();
+  this.roomIdInput_.addEventListener('input', this.onRoomIdInput_.bind(this), false);
+  this.roomRandomButton_.addEventListener('click', this.onRandomButton_.bind(this), false);
+  this.roomJoinButton_.addEventListener('click', this.onJoinButton_.bind(this), false);
+  
+  // Public callbacks. Keep it sorted.
+  this.onRoomSelected = null;
+  
+  this.recentlyUsedList_ = new RoomSelection.RecentlyUsedList(recentRoomsKey);
+  this.startBuildingRecentRoomList_();
+};
+
+RoomSelection.prototype.startBuildingRecentRoomList_ = function() {
+  this.recentlyUsedList_.getRecentRooms().then(this.buildRecentRoomList_.bind(this)).catch(function(error) {
+    trace('Error building recent rooms list: ' + error.message);
+  }.bind(this));
+};
+
+RoomSelection.prototype.buildRecentRoomList_ = function(recentRooms) {
+  for (var i = 0; i < recentRooms.length; ++i) {
+    // Create link in recent list
+    var li = document.createElement('li');
+    var href = document.createElement('a');
```

Comment:
They were originally actual navigation links in the web app. We can/should change this when we implement the upcoming UI changes.